test/integration/singlecluster/controller/jobs/rayjobAndRaycluster/rayjob_and_raycluster_controller_test.go

```diff
@@ -0,0 +1,355 @@
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	testingraycluster "sigs.k8s.io/kueue/pkg/util/testingjobs/raycluster"
+	testingrayjob "sigs.k8s.io/kueue/pkg/util/testingjobs/rayjob"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+const (
+	instanceKey = "cloud.provider.com/instance"
+)
+
+var _ = ginkgo.Describe("RayJob and RayCluster Ancestor Workload Handling", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+	// These tests verify that Kueue's FindAncestorJobManagedByKueue function correctly
+	// handles the parent-child relationship between RayJob and RayCluster.
+	//
+	// The expected behavior is:
+	// - If a RayCluster is owned by a RayJob that has a queue-name label, the RayCluster
+	//   should NOT create its own workload (even if it has a queue-name label).
+	// - Only the top-level job in the ownership chain should create a workload.
+	// - This prevents duplicate workloads and resource double-counting when KubeRay
+	//   propagates labels from RayJob to RayCluster.
+
+	ginkgo.BeforeAll(func() {
+		// Start both RayJob and RayCluster controllers
+		fwk.StartManager(ctx, cfg, managerAndSchedulerSetup(
+			jobframework.WithManageJobsWithoutQueueName(false)))
+	})
+	ginkgo.AfterAll(func() {
+		fwk.StopManager(ctx)
+	})
+
+	var (
+		ns           *corev1.Namespace
+		flavor       *kueue.ResourceFlavor
+		clusterQueue *kueue.ClusterQueue
+		localQueue   *kueue.LocalQueue
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = util.CreateNamespaceFromPrefixWithLog(ctx, k8sClient, "ray-toplevel-")
+
+		// Create test infrastructure
+		flavor = testing.MakeResourceFlavor("default").NodeLabel(instanceKey, "default").Obj()
+		util.MustCreate(ctx, k8sClient, flavor)
+
+		clusterQueue = testing.MakeClusterQueue("cluster-queue").
+			ResourceGroup(
+				*testing.MakeFlavorQuotas("default").Resource(corev1.ResourceCPU, "10").Obj(),
+			).
+			Obj()
+		util.MustCreate(ctx, k8sClient, clusterQueue)
+
+		localQueue = testing.MakeLocalQueue("local-queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+		util.MustCreate(ctx, k8sClient, localQueue)
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, localQueue, true)
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, clusterQueue, true)
+		util.ExpectObjectToBeDeleted(ctx, k8sClient, flavor, true)
+	})
+
+	ginkgo.It("Should NOT create a workload for RayCluster when owned by RayJob (both have queue-name labels)", func() {
+		ginkgo.By("1. Create a RayJob with queue label")
+		rayJob := testingrayjob.MakeJob("test-rayjob", ns.Name).
+			Queue(localQueue.Name).
+			RequestHead(corev1.ResourceCPU, "1").
+			RequestWorkerGroup(corev1.ResourceCPU, "1").
+			Obj()
+		util.MustCreate(ctx, k8sClient, rayJob)
+
+		ginkgo.By("2. Wait for RayJob to have a workload created")
+		var rayJobWorkload *kueue.Workload
+		gomega.Eventually(func() bool {
```

Comment:
```suggestion
		gomega.Eventually(func(g gomega.Gomega) {
```
Please use `g gomega.Gomega` instead of returning bool for consistency with other places.